### PR TITLE
Packaged apps views + DustAPI interface

### PR DESCRIPTION
**Pull request by spolu**
- Introduces a `Use` view awith `Use` / `Build` switch
- Routes role users to it
- Introduces DustAPI typescript library to interface with Production Dust
- Adds a public API route to list data sources

In upcoming PRs:
- ProdDustAppRegistry for each packaged app
- Implementation of Chat app v0 (removed from this PR to make it easy to review)

### CODE REVIEW

**Comment by gabhubert** [comment]
Nice! A few humble comments @spolu 
- **Apps**: I'd put "write" last and I'd name it "Notes" or "Drafts" (noun, like Answers and Alerts)
- **mode switch**: imho it's not super clear to me that it's an affordance and the arrows are a bit of a repeat under the docs arrows (the purple colour doesn't mean clickable in the text below for example.) 
  - thought 1: Should we scrap "view documentation" (which is really not central) for design partners and have the mode there? Maybe too much work to maintain given the many non-design-partner users? 
  - thought 2: could we try a visual where you just have admin || builder and user next to each other, one (the active one) in purple and the other one in grey (and have the same hover state as we do on the left for when an affordance is grey)

Screenshot 2023-05-05 at 15 13 38
- **Copy:**
  - Answers will tackle any internal question across your team's knowledge base, in a format adapted to your needs.
  - Alerts will stay on top of things in real time with updates crafted for you, on topics you choose, at the frequency you need.
  - Notes will let you create better content faster with the entire team's knowledge at your fingertips.
  - Across all: reach out at team@dust.tt with any ideas on what you'd like to see here.

**Comment by spolu** [comment]
Thanks! Will implement :+1:

**Comment by spolu** [comment]
Note: comments from @gabhubert addressed and copy integrated :+1:

File: front/components/use/MainTab.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      name: "Alerts",
+      href: `/w/${owner.sId}/u/alerts`,
+      icon: (
+        <BellAlertIcon
+          className="mr-2 mt-0.5 h-4 w-4 flex-shrink-0"
+          aria-hidden="true"
+        />
+      ),
+    },
+  ] as { name: string; href: string; icon: JSX.Element }[];
```
Comment (fontanierh) [inline front/components/use/MainTab.tsx:61]: nit: why `as {}[]` instead of `const tabs: {}[] =` ? The latter actually type-checks

**Review by fontanierh — APPROVED** [review]
LGTM

**Review by fontanierh — DISMISSED** [review]
LGTM

File: front/components/use/MainTab.tsx (hunk 1 above)
Comment (spolu) [reply front/components/use/MainTab.tsx:61]: me being a bad typescripter will address and merge :+1:

**Pull request merged**